pkg/backfill/config.go

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package backfill
+
+import (
+	"time"
+)
+
+type Config struct {
+	batchSize  int
+	batchDelay time.Duration
+	callbacks  []CallbackFn
+}
+
+const (
+	DefaultBatchSize int           = 1000
+	DefaultDelay     time.Duration = 0
+)
+
+type OptionFn func(*Config)
+
+func NewConfig(opts ...OptionFn) *Config {
+	c := &Config{
+		batchSize:  DefaultBatchSize,
+		batchDelay: DefaultDelay,
+		callbacks:  make([]CallbackFn, 0),
+	}
+
+	for _, opt := range opts {
+		opt(c)
+	}
+	return c
+}
+
+// WithBatchSize sets the batch size for the backfill operation.
+func WithBatchSize(batchSize int) OptionFn {
+	return func(o *Config) {
+		o.batchSize = batchSize
+	}
+}
+
+// WithBatchDelay sets the delay between batches for the backfill operation.
+func WithBatchDelay(delay time.Duration) OptionFn {
+	return func(o *Config) {
+		o.batchDelay = delay
+	}
+}
+
+// WithCallbacks sets the callbacks for the backfill operation.
+// Callbacks are invoked after each batch is processed.
+func WithCallbacks(cbs ...CallbackFn) OptionFn {
```

Comment:
Could remove this and change its only usage to `AddCallback` instead. Or remove `AddCallback` and update its usage to `WithCallbacks`. We don't need both methods IMO.